City-of-Helsinki/drupal-helfi-platform-config
# UHF-8650 Preparation for the HDS cookie banner module

# [UHF-8650](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8650)


## What was done


* Renamed the cookie policy to cookie settings.
* Added default JSON without any cookies to "site_settings". This will prevent the JSON validation bug when enabling site specific cookie settings.
* Changed the cookie policy page controller to cookie settings page controller.
* Converted the cookie banner attachment to a service.
* Added install hooks for the cookie banner module and uninstall hook for the helfi eu cookie compliance module.
* Added tests for the Cookie settings service.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-8650 drupal/hdbt:dev-UHF-8650 -W`
* Run `make drush-updb drush-cr`

## How to test


Check testing instructions from https://github.com/City-of-Helsinki/drupal-hdbt/pull/1083

## Other PRs


* https://github.com/City-of-Helsinki/drupal-hdbt/pull/1083


[UHF-8650]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8650?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
